src/crypto/hash.cpp

```diff
@@ -23,57 +26,93 @@ namespace crypto
     mbedtls_sha256_free(&ctx);
   }
 
-  class MBSha256HashImpl
+#ifdef HAVE_OPENSSL
```

Comment:
As you pointed out, we can get rid of those, we depend on a version of Open Enclave (>= 0.13) that guarantees this.